server/src/main/java/org/opensearch/action/main/MainResponse.java

```diff
@@ -136,6 +143,13 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    private String getVersionNumber() {
+        if (clientCompatibilityVersion != null && clientCompatibilityVersion.equals(Version.V_EMPTY) == false) {
```

Comment:
Alternatively, the need for this `if` tells me that it should have happened earlier our outside of here, aka either the input to `MainRespose` was the version we wanted to return, or `build.getVersionNumber()` should become `build.getResponseVersionNumber()` where all the logic happens. It's easier to unit test when the logic for the version number defaulting is self-contained.

Reply:
It seems the `Build` is concerned with the build information and probably should not be the place. But, perhaps moving this condition into `TransportMainAction` would make sense.

Reply:
Agree, I have removed the need for this by making the setting the default to the current build version.